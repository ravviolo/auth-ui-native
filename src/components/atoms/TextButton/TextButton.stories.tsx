import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { TextButton } from './TextButton';

storiesOf('TextButton', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <TextButton
      testID={text('testID', 'text-btn-test-id')}
      title={text('title', 'Button')}
      onPress={action('onPress')}
    />
  ));
