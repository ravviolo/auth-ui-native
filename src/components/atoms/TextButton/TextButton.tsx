import { StyleProp, Text, TextStyle, TouchableOpacity } from 'react-native';

export interface Props {
  title: string;
  testID: string;
  onPress: () => void;
  style?: StyleProp<TextStyle>;
}

export const TextButton = ({ onPress, title, testID, style }: Props) => {
  return (
    <TouchableOpacity testID={testID} onPress={onPress}>
      <Text style={style}>{title}</Text>
    </TouchableOpacity>
  );
};
