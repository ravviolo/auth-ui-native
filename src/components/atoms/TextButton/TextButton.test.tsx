/* eslint-disable react-native/no-inline-styles */
import { render, screen, fireEvent } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { TextButton } from './TextButton';

import type { Props } from './TextButton';

describe('TextButton', () => {
  const mockOnPress = jest.fn<Props['onPress'], []>();

  const props: Props = {
    title: 'Test',
    testID: 'text-btn-test-id',
    onPress: mockOnPress,
  };

  it('should render button with text received through props', () => {
    render(<TextButton {...props} />);

    const textBtn = screen.getByTestId('text-btn-test-id');

    expect(textBtn).toHaveTextContent('Test');
  });

  it('should handle events, run handler once', () => {
    render(<TextButton {...props} />);

    const textBtn = screen.getByTestId('text-btn-test-id');

    fireEvent.press(textBtn);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should apply additional styles received via props', () => {
    render(<TextButton {...props} style={{ textTransform: 'capitalize' }} />);

    const textBtn = screen.getByTestId('text-btn-test-id');

    expect(textBtn.children[0]).toHaveStyle({ textTransform: 'capitalize' });
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<TextButton {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
