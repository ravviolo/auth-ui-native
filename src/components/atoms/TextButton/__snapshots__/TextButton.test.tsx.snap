// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="text-btn-test-id"
>
  <Text>
    Test
  </Text>
</View>
`;
